src/routes/blog/index.svelte

```diff
@@ -1,21 +1,48 @@
 <script lang="ts" context="module">
   import type { LoadInput, LoadOutput } from '@sveltejs/kit/types/internal';
 
-  export async function load({ fetch }: LoadInput): Promise<LoadOutput> {
-    const response = await fetch(`/blog.json`);
-    return { props: { posts: await response.json() } };
+  export async function load({ url }: LoadInput): Promise<LoadOutput> {
+    const target = new URL('/blog.json', url);
+    const rawLabels = url.searchParams.get('l');
+    const selectedLabels = [];
+
+    if (rawLabels !== null && rawLabels.length > 0) {
```

Comment:
Nit: Simplification of conditional statement. 
```suggestion
    if (rawLabels?.length > 0) {
```